lib/queue.rb

```diff
@@ -1,31 +1,48 @@
 class Queue
-
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = -1
+    @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+    end
+    if @front == @back
+      raise ArgumentError, "QUEUE IS FULL. PLEASE STOP STUFFING IT."
+    end
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
+    if @store[@front].nil?
+      @front = (@front + 1) % @store.length
+    end
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if (size == 0)
+      return nil
+    end
+    temp_pop = @store[@front]
+    @store[@front] = nil
+    @front = (@front + 1) % @store.length
+    return temp_pop
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return @store.compact.size
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    return(size == 0)
   end
 
   def to_s
-    return @store.to_s
+    return @store[@front...@back].to_s
```

Comment:
What about when the back wraps around the array this only works when @front < @back